cylc/flow/config_schema.py

```diff
@@ -0,0 +1,660 @@
+            'required run mode': [
+                VDR.V_STRING,
+                '', 'live', 'simulation', 'dummy-local', 'dummy'],
+            'allow task failures': [VDR.V_BOOLEAN],
+            'expected task failures': [VDR.V_STRING_LIST],
+            'live mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'dummy mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'dummy-local mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+            'simulation mode suite timeout': [
+                VDR.V_INTERVAL, DurationFloat(60)],
+        },
+        'authorization': {
+            # Allow owners to grant public shutdown rights at the most, not
+            # full control.
+            'public': (
+                [VDR.V_STRING, '']
+                + [level.name.lower().replace('_', '-') for level in [
+                    Priv.IDENTITY, Priv.DESCRIPTION, Priv.STATE_TOTALS,
+                    Priv.READ, Priv.SHUTDOWN]])
+        },
+    },
+    'document viewers': {
+        'html': [VDR.V_STRING, 'firefox'],
+    },
+    'documentation': {
+        'local': [VDR.V_STRING, ''],
+        'online': [VDR.V_STRING,
+                   'http://cylc.github.io/doc/built-sphinx/index.html'],
+        'cylc homepage': [VDR.V_STRING, 'http://cylc.github.io/'],
+    },
+    'editors': {
+        'terminal': [VDR.V_STRING, 'vim'],
+        'gui': [VDR.V_STRING, 'gvim -f'],
+    },
+    'job platforms': {
+        'default platform': {
+            'run directory': [VDR.V_STRING, '$HOME/cylc-run'],
+            'work directory': [VDR.V_STRING, '$HOME/cylc-run'],
+            'task communication method': [
+                VDR.V_STRING, 'default', 'ssh', 'poll'],
+            'submission polling intervals': [VDR.V_INTERVAL_LIST],
+            'execution polling intervals': [VDR.V_INTERVAL_LIST],
+            'scp command': [
+                VDR.V_STRING, 'scp -oBatchMode=yes -oConnectTimeout=10'],
+            'ssh command': [
+                VDR.V_STRING, 'ssh -oBatchMode=yes -oConnectTimeout=10'],
+            'use login shell': [VDR.V_BOOLEAN, True],
+            'login hosts': [VDR.V_STRING, ''],
+            'batch system': [VDR.V_STRING, ''],
+            'cylc executable': [VDR.V_STRING, 'cylc'],
+            'global init-script': [VDR.V_STRING],
+            'copyable environment variables': [VDR.V_STRING_LIST],
+            'retrieve job logs': [VDR.V_BOOLEAN],
+            'retrieve job logs command': [VDR.V_STRING, 'rsync -a'],
+            'retrieve job logs max size': [VDR.V_STRING],
+            'retrieve job logs retry delays': [VDR.V_INTERVAL_LIST],
+            'task event handler retry delays': [VDR.V_INTERVAL_LIST],
+            'tail command template': [
+                VDR.V_STRING, 'tail -n +1 -F %(filename)s'],
+            'batch systems': {
+                '__MANY__': {
```

Comment:
There will be a single batch system per job platform, so you should be able to flatten this section.

Reply:
I did rather wonder!